src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesService.java

```diff
@@ -54,6 +59,33 @@ public OverseasEntitiesService(OverseasEntitySubmissionsRepository overseasEntit
         this.dateTimeNowSupplier = dateTimeNowSupplier;
     }
 
+    public SubmissionType getSubmissionType(String overseasEntityId) throws SubmissionNotFoundException {
+        if (!isROEUpdateEnabled) {
+            return SubmissionType.REGISTRATION;
+        }
+        Optional<OverseasEntitySubmissionDto> submissionOpt = getOverseasEntitySubmission(
+                overseasEntityId);
+        if (submissionOpt.isEmpty()) {
+            var errorMessage = "Can not determine submission type";
+            Map<String, Object> logMap = new HashMap<>();
+            logMap.put(OVERSEAS_ENTITY_ID_KEY, overseasEntityId);
+            var error = new SubmissionNotFoundException(errorMessage);
+            ApiLogger.error(errorMessage, error, logMap);
```

Comment:
For the calls to ApiLogger, can you use the xxxContext versions of the methods and pass the requestID in as the context? It should allow us to link the log messages together more easily when following a request through the logs.